TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -224,5 +230,78 @@ public static CompletionItem CreateCompletionItemForIndex(Compiler.CodeElements.
             var display = string.Format("{0} (from {1})", index.Name, variable.Name);
             return new CompletionItem(display) {insertText = index.Name, kind = CompletionItemKind.Variable};
         }
+
+        public static Case GetTextCase(string tokenText)
+        {
+            if (string.IsNullOrEmpty(tokenText)) return Case.Lower;
+            // check if upper case
+            bool isUpper = true;
+            foreach (char c in tokenText)
+            {
+                if (char.IsLower(c))
+                {
+                    isUpper = false;
+                    break;
+                }
+            }
+
+            if (isUpper) return Case.Upper;
+            // check if Camel case; ex. CamlSample
+            bool isCamel = false;
+            int iUpperMem = -1, count = tokenText.Length;
+            for (int i = 0; i < count; i++)
+            {
+                if (i == 0)
+                {
+                    // first letter
+                    if (char.IsLower(tokenText[i])) break;
+                    isCamel = true;
+                }
+                else
+                {
+                    if (char.IsUpper(tokenText[i]))
+                    {
+                        if (iUpperMem == (i - 1))
+                        {
+                            // 2 upper case characters one behind the other
+                            isCamel = false;
+                            break;
+                        }
+                        iUpperMem = i;
+                    }
+                }
+            }
+
+            if (isCamel && iUpperMem == count - 1) isCamel = false; // last char is upper
+            return (isCamel) ? Case.Camel : Case.Lower;
+        }
+
+        public static Dictionary<string, string> GetParamsWithCase(Case textCase)
+        {
+            if (textCase == Case.Upper) return new Dictionary<string, string>()
+            {
+                { "INPUT", "INPUT"},
```

Comment:
Can you use constant for keys: "INPUT", "IN-OUT" and "OUTPUT" ?

Reply:
fixed by using ParameterDescription.PassingTypes for keys
https://github.com/TypeCobolTeam/TypeCobol/tree/6d2502c2ac02e718a765ac3e549933b59b6ebb69